File: demo/src/main/java/com/example/demo/api/HelloRestController.java
package com.example.demo.api;

import com.example.demo.response.Greeting;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloRestController {
  @GetMapping("/hello")
  public Greeting hello(
    @RequestParam(value = "name", defaultValue = "World") final String name
  ) {
    return new Greeting(1, "hello" + name);
  }
}

File: demokotlin/src/main/kotlin/com/example/demokotlin/api/HelloRestController.kt
package com.example.demokotlin.api

import com.example.demokotlin.response.Greeting
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.GetMapping


@RestController
class HelloRestController {
  @GetMapping("/hello")
  fun hello(@RequestParam(value="name", defaultValue = "world")name : String):Greeting {
    val xx = Greeting(1, name)
    return xx
  }
}
